portfolio/src/main/java/com/google/sps/servlets/EditDataServlet.java

```diff
@@ -92,30 +92,65 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             logger.warning("Could not convert User's likes list to ArrayList<Long>"); 
           }
 
-          // check to make sure user's list of liked comments 
-          // doesn't have this comment inside of it
-          if (!userLikes.contains(commID)) {
+          ArrayList<Long> userDislikes = null;
+          Object objUserDislikes = user.getProperty("dislikes");
+          if (objUserDislikes instanceof ArrayList) {
+            userDislikes = (ArrayList<Long>) objUserDislikes;
+          } else {
+            logger.warning("Could not convert User's dislikes list to ArrayList<Long>"); 
+          }
           
+          if (userLikes.contains(commID)) {
+            if (type.equals("like")) {
+              // unlike
+              Object likes = comment.getProperty("likes");
+              if (likes instanceof Long) {
+                comment.setProperty("likes", ((long) likes) - 1);
+              } else {
+                logger.warning("Could not convert comment's likes to long"); 
+              }
+              userLikes.remove(commID);
+            } else if (type.equals("dislike")) {
+              // invalid
+            }
```

Comment:
No need to add this case because the code does not do anything.

Maybe the like/dislike handling logic can also be simplified a bit:

```
if (type.equals("like")) {
    if (userLikes.contains(commID)) {
        // do unlike
    } else if (!userDislikes.contains(commID)) {
        // do like
    }
} else if (type.equals("dislike")) {
    if (userDislikes.contains(commID)) {
        // do un-dislike
    } else if (!userLikes.contains(commID)) {
        // do dislike
    }
}
```

Reply:
Got it!